src/codegen/function.rs

```diff
@@ -226,10 +229,40 @@ pub fn declaration_futures(env: &Env, analysis: &analysis::functions::Info) -> S
     )
 }
 
-pub fn bounds(bounds: &Bounds, skip: &[char], async: bool) -> String {
+pub fn bound_to_string(bound: &Bound, async: bool) -> String {
+    use analysis::bounds::BoundType::*;
+
+    match bound.bound_type {
+        NoWrapper => {
+            format!("{}: {}", bound.alias, bound.type_str)
+        }
+        IsA(Some(lifetime)) => {
+            format!("{}: IsA<{}> + {}", bound.alias, bound.type_str, if async { "Clone + 'static".into() } else { format!("'{}", lifetime) })
+        }
+        IsA(None) => format!("{}: IsA<{}>{}", bound.alias, bound.type_str, if async { " + Clone + 'static" } else { "" }),
+        // This case should normally never happened
+        AsRef(Some(lifetime)) => {
+            format!("{}: AsRef<{}> + '{}", bound.alias, bound.type_str, lifetime)
+        }
+        AsRef(None) => format!("{}: AsRef<{}>", bound.alias, bound.type_str),
+        Into(Some(l), _) => format!("{}: Into<Option<&'{} {}>>",
+                                    bound.alias,
+                                    l,
+                                    bound.type_str),
+        Into(None, _) => format!("{}: Into<Option<{}>>", bound.alias, bound.type_str),
+    }
+}
+
+pub fn bounds(
+    bounds: &Bounds,
+    skip: &[char],
+    async: bool,
+    filter_hack: bool,
```

Comment:
What's a `filter_hack`? Should probably be either considered not a hack before merging, or cleaned up

Reply:
The important part is "filter" here. It filters out types with the `hack` field set to `true`. Once I'll have renamed the `hack` field, I'll update this field's name too.

Reply:
Ah it's filtering hacks, not a hack for filtering :)